File: src/csharp/SlimShader/Chunks/Shex/Tokens/OperandIndex.cs
namespace SlimShader.Chunks.Shex.Tokens
{
	public class OperandIndex
	{
		public OperandIndexRepresentation Representation { get; set; }
		public ulong Value { get; set; }
		public Operand Register { get; set; }

		public override string ToString()
		{
			string result = string.Empty;
			if (Register != null)
				result += Register + " + ";
			result += Value;
			return result;
		}
	}
}

File: src/cpp/Source/Chunks/Shex/Tokens/OperandIndex.cpp
#include "PCH.h"
#include "OperandIndex.h"

#include "Operand.h"

using namespace std;
using namespace SlimShader;

std::ostream& SlimShader::operator<<(std::ostream& out, const OperandIndex& value)
{
	if (value.Register != nullptr)
		out << *value.Register << " + ";
	out << value.Value;
	return out;
}